synapse/federation/sender/per_destination_queue.py

```diff
@@ -192,9 +194,7 @@ def attempt_new_transaction(self) -> None:
             logger.debug("TX [%s] Transaction already in progress", self._destination)
             return
 
-        if not self._federation_shard_config.should_send_to(
-            self._instance_name, self._destination
-        ):
+        if self._should_send_on_this_instance:
```

Comment:
I think you're missing a `not` here:

```suggestion
        if not self._should_send_on_this_instance:
```